frontend/register.html

```diff
@@ -0,0 +1,145 @@
+            </select>
+            <div id="professionalFields">
+                <input type="text" id="profession" placeholder="Profession">
+                <input type="text" id="skills" placeholder="Skills (comma-separated)">
+                <input type="number" id="experience" placeholder="Years of Experience">
+                <input type="text" id="location" placeholder="Location">
+            </div>
+            <button type="submit">Register</button>
+        </form>
+        <div class="toggle">
+            Already have an account? <a href="signin.html">Sign In</a>
+        </div>
+    </div>
+
+    <script>
+        const API_URL = 'http://localhost:5000/api'; // Update this with your actual backend URL
+
+        document.getElementById('userType').addEventListener('change', function() {
+            const professionalFields = document.getElementById('professionalFields');
+            professionalFields.style.display = this.value === 'professional' ? 'block' : 'none';
+        });
+
+        document.getElementById('registerForm').addEventListener('submit', async function(e) {
+            e.preventDefault();
+            
+            const userData = {
+                name: document.getElementById('registerName').value,
+                email: document.getElementById('registerEmail').value,
+                password: document.getElementById('registerPassword').value,
+                userType: document.getElementById('userType').value,
+            };
+
+            if (userData.password !== document.getElementById('confirmPassword').value) {
+                alert('Passwords do not match');
+                return;
+            }
+
+            if (userData.userType === 'professional') {
+                userData.profession = document.getElementById('profession').value;
+                userData.skills = document.getElementById('skills').value.split(',').map(skill => skill.trim());
+                userData.experience = document.getElementById('experience').value;
+                userData.location = document.getElementById('location').value;
+            }
+
+            try {
+                const response = await fetch(`${API_URL}/auth/register`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify(userData),
+                });
+
+                if (response.ok) {
+                    alert('Registered successfully! Please sign in.');
+                    window.location.href = 'signin.html';
+                } else {
+                    const errorData = await response.text();
+                    alert(`Registration failed: ${errorData}`);
+                }
+            } catch (error) {
+                console.error('Error:', error);
+                alert('An error occurred. Please try again.');
+            }
+        });
+    </script>
+</body>
+</html>
```

Comment:
Code is well readable but here are few points
 Add password strength validation, sanitize user inputs to prevent XSS.
User Experience: Add real-time validation and provide better feedback for incorrect form inputs.
Code Structure: Refactor the JavaScript code into reusable modules for better maintainability.
Error Handling: Improve error messaging by checking specific failure conditions.